src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -0,0 +1,589 @@
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateDouble(double value)
+        {
+            if (double.IsPositiveInfinity(value) || double.IsNegativeInfinity(value) || double.IsNaN(value))
+                ThrowHelper.ThrowArgumentException_ValueNotSupported();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateSingle(float value)
+        {
+            if (float.IsPositiveInfinity(value) || float.IsNegativeInfinity(value) || float.IsNaN(value))
+                ThrowHelper.ThrowArgumentException_ValueNotSupported();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateProperty(ref ReadOnlySpan<char> propertyName)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize)
+                ThrowHelper.ThrowArgumentException_PropertyNameTooLarge(propertyName.Length);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateValue(ref ReadOnlySpan<char> value)
+        {
+            if (value.Length > JsonConstants.MaxCharacterTokenSize)
+                ThrowHelper.ThrowArgumentException_ValueTooLarge(value.Length);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidatePropertyAndValue(ref ReadOnlySpan<char> propertyName, ref ReadOnlySpan<byte> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize || value.Length > JsonConstants.MaxTokenSize)
+                ThrowHelper.ThrowArgumentException(propertyName, value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidatePropertyAndValue(ref ReadOnlySpan<byte> propertyName, ref ReadOnlySpan<char> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxTokenSize || value.Length > JsonConstants.MaxCharacterTokenSize)
+                ThrowHelper.ThrowArgumentException(propertyName, value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidatePropertyAndValue(ref ReadOnlySpan<byte> propertyName, ref ReadOnlySpan<byte> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxTokenSize || value.Length > JsonConstants.MaxTokenSize)
+                ThrowHelper.ThrowArgumentException(propertyName, value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidatePropertyAndValue(ref ReadOnlySpan<char> propertyName, ref ReadOnlySpan<char> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize || value.Length > JsonConstants.MaxCharacterTokenSize)
+                ThrowHelper.ThrowArgumentException(propertyName, value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountDigits(ulong value)
```

Comment:
Can't the project just include https://github.com/dotnet/corefx/blob/109f12f78a0b70d9739d11fa351b740ff77e3695/src/Common/src/CoreLib/System/Buffers/Text/FormattingHelpers.CountDigits.cs and use CountDigits from there rather than duplicating the code?